### DISCUSSION TITLE
Is it possible to get alerts for lack of updates?

### DISCUSSION THREAD

**Discussion by elmehalawi** [MaterializeInc/materialize#6551] [Q&A]
I'm interested in getting alerts when a user hasn't logged in in so many days. Is this possible using materialize?

**Comment by frankmcsherry** [comment]
Great question! This should be possible to express with [temporal filters](https://materialize.com/temporal-filters/). They allow you to express a constraint on the results that are presented which involves the "logical time", which is the maintained time as the query runs. For example, you could determine the most recent log in time (a `max`) and then wrap that with a 
```sql
having mz_logical_timestamp() > last_login_ms + <10 days in ms>
```
The timestamps here are all "milliseconds since the 1970 epoch", which is how your data are most likely timestamped (this does depend on how you are introducing data to Materialize, and you would know if you were using a specifically timestamped source).
